tools/depmod.c

```diff
@@ -389,37 +462,45 @@ static uint32_t index_write__node(const struct index_node *node, FILE *out)
 			fputs(v->value, out);
 			fputc('\0', out);
 		}
-		offset |= INDEX_NODE_VALUES;
 	}
 
-	return offset;
+	/* Now write children */
+	if (child_count) {
+		int i;
+
+		offset += size;
+		for (i = 0; i < child_count; i++) {
+			struct index_node *child;
+
+			child = node->children[node->first + i];
+			if (child != NULL) {
+				uint32_t s, t;
+				index_get_size(child, &s, &t);
+				index_write__node(child, out, offset);
```

Comment:
Oh, indeed. That's nicer. I'll integrate it.

Reply:
I would have skipped these ones and integrate it. Checked as well if such init statements already exist and since they do not, I won't introduce them with this PR.

Reply:
Oh I see, a removal of malloc. That's clever. I'll add it as a separate commit to this PR